lib/ssrf_filter/ssrf_filter.rb

```diff
@@ -191,17 +191,17 @@ def self.fetch_once(uri, ip, verb, options, &block)
 
     with_forced_hostname(hostname, ip) do
       ::Net::HTTP.start(uri.hostname, uri.port, **http_options) do |http|
-        http.request(request) do |response|
-          case response
-          when ::Net::HTTPRedirection
-            url = response['location']
-            # Handle relative redirects
-            url = "#{uri.scheme}://#{hostname}:#{uri.port}#{url}" if url.start_with?('/')
-            return nil, url
-          else
-            block&.call(response)
-            return response, nil
```

Comment:
This use of using `return` within a block is the cause of the issue. It causes the execution to jump out from the method `#fetch_once` (not from this block), skipping [post-processing of body read](https://github.com/ruby/net-http/blob/65aed40f350f8d13f727221fb391c677688cb705/lib/net/http/response.rb#L186) by net-http. So the result is attempting to read the body afterwards, when `@socket` already cleared up with `nil`.